typedload/dataloader.py

```diff
@@ -233,7 +239,18 @@ def load(self, value: Any, type_: Type[T]) -> T:
                 self.frefs[tname] = type_
 
         func = self.handlers[index][1]
-        return func(self, value, type_)
+        try:
+            return func(self, value, type_)
+        except TypedloadException as e:
+            e.trace.insert(0, TraceItem(value, type_, annotation))
+            raise e
+        except Exception as e:
+            # This is a bug.
+            raise TypedloadException(
+                '==== SOFTWARE BUG ====: %s. Non TypedloadException encountered' % e,
+                type_=type_,
+                value=value,
+            )
```

Comment:
Wouldn't an assert make sense here instead? Or a different exception type.

Reply:
Yeah you're right